internal/keys/hasher.go

```diff
@@ -45,16 +45,16 @@ func (t tupleKeysHasher) Append(h hasher) error {
 	sortedTupleKeys := append([]*openfgav1.TupleKey(nil), t.tupleKeys...) // Copy input to avoid mutating it
 
 	sort.SliceStable(sortedTupleKeys, func(i, j int) bool {
-		if sortedTupleKeys[i].Object > sortedTupleKeys[j].Object {
-			return false
+		if sortedTupleKeys[i].Object != sortedTupleKeys[j].Object {
+			return sortedTupleKeys[i].Object < sortedTupleKeys[j].Object
```

Comment:
For a followup PR: do you mind checking the rest of the codebase for uses of `sort.SliceStable`? We may have this problem elsewhere too